library/src/main/java/com/google/android/material/motion/runtime/Scheduler.java

```diff
@@ -172,6 +172,38 @@ public void addPlan(Plan plan, Object target) {
     getTargetScope(target).commitPlan(planInfo);
   }
 
+  /**
+   * Adds a {@link NamedPlan} to this scheduler.
+   * @param plan the {@link NamedPlan} to add to the scheduler.
+   * @param name the name by which this plan can be identified.
+   * @param target the target on which the plan will operate.
+   */
+  public void addNamedPlan(NamedPlan plan, String name, Object target) {
+    if (name == null || name.length() == 0) {
+      throw new AssertionError("A NamedPlan must have a name with more than zero characters");
+    }
+    PlanInfo planInfo = new PlanInfo();
+    planInfo.name = name;
+    planInfo.target = target;
+    planInfo.plan = plan.clone();
+    getTargetScope(target).commitAddNamedPlan(planInfo);
+  }
+
+  /**
+   * Removes a {@link NamedPlan} from this scheduler.
+   * @param name the name by which the named plan can be identified.
+   * @param target the target on which the named plan was added.
+   */
+  public void removePlanNamed(String name, Object target) {
+    if (name == null || name.length() == 0) {
+      throw new AssertionError("A NamedPlan must have a name with more than zero characters");
+    }
+    PlanInfo planInfo = new PlanInfo();
+    planInfo.name = name;
+    planInfo.target = target;
+    getTargetScope(target).commitRemoveNamedPlan(planInfo);
```

Comment:
Hmmm. I've made a change to pass along the name, plan and target in some of these API calls which basically allows me to not create a `PlanInfo` object in the `Scheduler`. Not sure if this is the idea you had in mind when you suggested this change, but that's where I've ended up. If we're thinking about moving away from `Transactions`, I think moving away from `PlanInfo` objects mightn't be a bad idea at this stage?